plugins/apitracing/src/lib/Tracer.cpp

```diff
@@ -70,10 +69,10 @@ namespace ApiTracing
     }
 
     void Tracer::injectHooks(const ActiveProcessInformation& processInformation,
-                             const std::optional<ProcessTracingConfig>& processTracingConfig)
+                             const std::optional<TracingProfile>& processTracingProfile)
     {
         auto introspectionAPI = pluginInterface->getIntrospectionAPI();
-        for (const auto& moduleHookTarget : processTracingConfig->profile.modules)
+        for (const auto& moduleHookTarget : processTracingProfile->modules)
```

Comment:
Optionals should be checked whether they contain something before being accessed. Since you've already done that in the parent function I would suggest changing the signature of the function by removing the optional.